crates/ty_python_semantic/src/types/constraints.rs

```diff
@@ -1037,7 +1036,8 @@ impl<'db> Node<'db> {
             Node::Interior(_) => {}
         }
 
-        let mut typevars = FxHashSet::default();
+        // We should use `FxIndexSet` here since `BoundTypeVarInstance::{valid, required}_specializations` is query-dependent.
+        let mut typevars = FxIndexSet::default();
```

Comment:
We should probably use `FxIndexSet` here.
The rest of this module seems okay to use unstable iterators, unless I'm overlooking something.